src/value/owned.rs

```diff
@@ -357,110 +350,169 @@ impl Element for OwnedElement {
+        case::f64(
+            OwnedValue::Float(16.0),
+            IonType::Float,
+            AsF64,
+            &|e: &OwnedElement| assert_eq!(Some(16.0), e.as_f64())
+        ),
+        case::decimal(
+            OwnedValue::Decimal(Decimal::new(8, 3)),
+            IonType::Decimal,
+            AsDecimal,
+            &|e: &OwnedElement| assert_eq!(Some(&Decimal::new(80, 2)), e.as_decimal())
+        ),
+        case::timestamp(
+            OwnedValue::Timestamp(mk_ts("2014-10-16T12:01:00-00:00")),
+            IonType::Timestamp,
+            AsTimestamp,
+            &|e: &OwnedElement| assert_eq!(Some(&mk_ts("2014-10-16T12:01:00+00:00")), e.as_timestamp())
+        ),
+        case::str(
+            OwnedValue::String("hello".into()),
+            IonType::String,
+            AsStr,
+            &|e: &OwnedElement| assert_eq!(Some("hello"), e.as_str())
+        ),
+        case::sym_text(
+            OwnedValue::Symbol("hello".into()),
+            IonType::Symbol,
+            vec![AsStr, AsSym],
+            &|e: &OwnedElement| {
+                assert_eq!(Some("hello"), e.as_str());
+                assert_eq!(Some("hello"), e.as_sym().unwrap().text());
+            }
+        ),
+        // TODO more symbol token tests (without text)
+        case::blob(
+            OwnedValue::Blob("world".as_bytes().into()),
+            IonType::Blob,
+            AsBytes,
+            &|e: &OwnedElement| {
+                assert_eq!(Some("world".as_bytes()), e.as_bytes());
+            }
+        ),
+        case::clob(
+            OwnedValue::Clob("goodbye".as_bytes().into()),
+            IonType::Clob,
+            AsBytes,
+            &|e: &OwnedElement| {
+                assert_eq!(Some("goodbye".as_bytes()), e.as_bytes());
+            }
+        ),
+    )]
+    fn owned_element_accessors<O: IntoIterator<Item = ElemOp>>(
+        val: OwnedValue,
+        ion_type: IonType,
+        valid_ops_iter: O,
+        op_assert: &ElemAssertFunc,
+    ) {
+        // table of negative assertions for each operation
+        let neg_table: Vec<(ElemOp, &ElemAssertFunc)> = vec![
+            (IsNull, &|e| assert_eq!(false, e.is_null())),
+            (AsBool, &|e| assert_eq!(None, e.as_bool())),
+            (AsAnyInt, &|e| {
+                assert_eq!(None, e.as_any_int());
+                assert_eq!(None, e.as_i64());
+                assert_eq!(None, e.as_big_int());
+            }),
+            (AsF64, &|e| assert_eq!(None, e.as_f64())),
+            (AsDecimal, &|e| assert_eq!(None, e.as_decimal())),
+            (AsTimestamp, &|e| assert_eq!(None, e.as_timestamp())),
+            (AsStr, &|e| assert_eq!(None, e.as_str())),
+            (AsSym, &|e| assert_eq!(true, e.as_sym().is_none())),
+            (AsBytes, &|e| assert_eq!(None, e.as_bytes())),
+            (AsSequence, &|e| assert_eq!(true, e.as_sequence().is_none())),
+            (AsStruct, &|e| assert_eq!(true, e.as_struct().is_none())),
+        ];
```

Comment:
One way to think about this is a kind of "vtable" for all the operators and the negative assertion, we then "override" with the input to the fixture.